.github/workflows/python-app.yml

```
@@ -21,5 +21,13 @@ jobs:
         python-version: "3.10"
     - name: Install gepetuto and its dependencies
       run: pipx install -e .
+    - name: Install isort
+      run: pipx install isort
+    - name: Install black
+      run: pipx install black
+    - name: Install ruff
+      run: pipx install ruff
+    - name: Install nbconvert
+      run: pip install nbconvert
```

```suggestion
      run: python -m pip install -U pip && python -m pip install .
```
At this point I guess we can directly use pip. At least in the CI. This will directly export isort black and ruff, plus nbconvert will be provided by jupyterlab. 
Poetry is still here to guarantee that all those projects don't have incompatible dependencies.

---

Also we won't have to update this list in this hidden file if the list ever change in pyproject.toml